src/network/networkd-address.c

```diff
@@ -53,15 +53,21 @@ int address_new(Address **ret) {
         return 0;
 }
 
-int address_new_static(Network *network, unsigned section, Address **ret) {
+int address_new_static(Network *network, const char *filename, unsigned section_line, Address **ret) {
+        _cleanup_network_config_section_free_ NetworkConfigSection *n = NULL;
         _cleanup_address_free_ Address *address = NULL;
         int r;
 
         assert(network);
+        assert(filename);
         assert(ret);
 
-        if (section) {
-                address = hashmap_get(network->addresses_by_section, UINT_TO_PTR(section));
+        if (section_line) {
+                r = network_config_section_new(filename, section_line, &n);
+                if (r < 0)
+                        return r;
+
+                address = hashmap_get(network->addresses_by_section, n);
```

Comment:
I think a certain optimization would make some sense here: as long as we only lookup entries in the hashmap (instead of storing some), it would make a ton of sense to construct a `NetworkConfigSection` structure on the fly using C99 structure literals:

```c
address = hashmap_get(network->addresses_by_section, &(NetworkConfigSection) { .filaname = filename, .line = section_line });
```

of course, as soon as we actually want to add an entry we need the memory allocation anyway, but let's avoid that for the common case.